packages/client/src/object/convertWireToOsdkObjects.ts

```diff
@@ -64,6 +122,13 @@ export function convertWireToOsdkObjects<
   for (const obj of objs) {
     Object.setPrototypeOf(obj, proto);
 
+    Object.defineProperty(obj, OriginClient, {
```

Comment:
I considered pushing this into the proto but then it made the caching semantics of `createCachedOntologyTransform` worse. Right now it caches on `===` ObjectTypeDefinition, which actually caches pretty well across client recreations and such.